pkg/accounts/credentials.go

```diff
@@ -0,0 +1,94 @@
+package accounts
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+
+	"github.com/cozy/cozy-stack/pkg/config"
+)
+
+const cipherHeader = "rsa"
+const plainPrefixLen = 4
+
+var (
+	errCannotDecrypt  = errors.New("accounts: cannot decrypt credentials")
+	errCannotEncrypt  = errors.New("accounts: cannot encrypt credentials")
+	errBadCredentials = errors.New("accounts: bad credentials")
+)
+
+// EncryptCredentials takes a login / password and encrypts their values using
+// the vault public key.
+func EncryptCredentials(login, password string) ([]byte, error) {
+	publicKey := config.GetVault().CredentialsPublicKey()
+	if publicKey == nil {
+		return nil, errCannotEncrypt
+	}
+
+	loginLen := len(login)
+
+	// make a buffer containing the length of the login in bigendian over 4
+	// bytes, followed by the login and password contatenated.
+	creds := make([]byte, plainPrefixLen+loginLen+len(password))
+
+	// put the length of login in the first 4 bytes
+	binary.BigEndian.PutUint32(creds[0:], uint32(loginLen))
+
+	// copy the concatenation of login + password in the end
+	copy(creds[plainPrefixLen:], login)
+	copy(creds[plainPrefixLen+loginLen:], password)
```

Comment:
Can you add a test where you encrypt and then decrypt a login+password where the login has some not-ascii characters?

It looks like `loginLen` is the number of runes of login, but `creds` is a slice of bytes and it should be the number of bytes of login that should be used to compute the offet. Maybe I'm missing something, so a test would be a nice way to be sure that we are doing this correctly.

Reply:
`loginLen` is actually the [number of bytes in the string](https://golang.org/pkg/builtin/#len). The number of runes would be returned by the method [RuneCountInString](https://golang.org/pkg/unicode/utf8/#RuneCountInString).

However, you're right and tests are on their way :)